slackbridge/bots.py

```diff
@@ -70,6 +81,27 @@ def _change_presence(self, user_bot):
             elif self.raw_message['presence'] == 'active':
                 user_bot.back()
 
+        def _post_to_imgur(self, user_bot, file_data):
+            ext = file_data['filetype']
+            if ext != 'png' and ext != 'jpg':
+                return
+
+            auth = {'Authorization': 'Bearer {}'.format(configs.slack_token)}
+            r = requests.get(file_data['url_private'],
+                             headers=auth, stream=True)
+            if r.status_code != 200:
+                return
+
+            with tempfile.NamedTemporaryFile('wb', suffix='.' + ext) as tempf:
+                r.raw.decode_content = True
+                shutil.copyfileobj(r.raw, tempf)
+                image = self.imgur.upload_image(tempf.name,
```

Comment:
Does the Imgur client throw any exceptions if the upload fails?

Reply:
I looked, and it uses the requests ```raise_for_status``` method under the hood if the upload fails. We could just catch a generic exception because we don't really care why it fails, log that it didn't work, and break out of that event call without posting anything.